### DISCUSSION TITLE
Zod coerce and form state value

### DISCUSSION THREAD

**Discussion by maurer2** [react-hook-form/react-hook-form#10530] [Q&A] [answered]
Hello, I noticed something unusual with react hook form and the `coerce` frunction from zod, when using the the coerce function with input fields (controlled and uncontrolled).

I have this schema (codesandbox: https://codesandbox.io/s/react-hook-form-zod-forked-djzcny?file=/src/App.tsx:153-392):

```ts
const schema = z.object({
  numberField: z.coerce
    .number({invalid_type_error: "Type error"})
    .min(1, { message: "Min value error" })
    .max(100, { message: "Max value error" })
});
```

the initial state

```ts
const initialFormState = {
  numberField: 1 // type number
};
```

and this component

```ts
export const App: FC = () => {
  const { register, handleSubmit, errors, watch } = useForm({
    resolver: zodResolver(schema),
    defaultValues: initialFormState
  });

  // outputs "number" on submit
  const onSubmit = (values: any) => {
    console.log("onSubmit:", typeof values?.numberField);
  };

  // outputs "string" after entering a value
  console.log("onWatch:", typeof watch("numberField"));

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>
        <span>numberField</span>
        <input name="numberField" ref={register} />
        <p>{errors.numberField?.message}</p>
      </label>
      <button type="submit">Submit</button>
    </form>
  );
};

```

When I log the type of numberField, I get "number" as the initial and correct value. When I enter a valid value afterwards, it keeps outputting "string" even tho the coerce-function should convert the field and only store a number value in the store.
I only seem to be getting the correct type in the onSubmit function.
Any ideas why this is happening?

Thanks

**Comment by leapful** [comment, marked as answer]
@maurer2 

`useWatch` hook get original value directly from input element, there is no schema validation execution on `useWatch` therefore the type of value is not coerce.

You need to cast original value by using schema manually to achieve the expected type value.
